### PR TITLE
RFC Forecast Reservoirs boolean variable for hydro.namelist

### LINKED ISSUE #411: Namelist option for RFC Forecast Reservoirs needed
A hydro.namelist option for RFC Forecast Reservoirs is needed.

## Expected Behavior
hydro.namelist will have an input variable for reservoir_rfc_forecasts that could be set to True/False in order to activate the the Reservoir sub-module that reads in RFC forecast time series files.

## Current Behavior
There is currently no option in namelist for RFC forecast reservoirs.

## Possible Solution
@jdmattern-noaa will submit a pull request with these fixes.

### PR DESCRIPTION
**Pull request by jdmattern-noaa**
TYPE: new feature
KEYWORDS: RFC Forecast Reservoirs, hydro.namelist, boolean
SOURCE: David Mattern - NOAA/OWP, @jdmattern-noaa 

DESCRIPTION OF CHANGES: hydro.namelist will have an input variable for reservoir_rfc_forecasts that could be set to True/False in order to activate the the Reservoir sub-module that reads in RFC forecast time series files. There is also a variable 'reservoir_not_levelpool' that is set to True whenever any type of reservoir other than levelpool is set to True. This variable is passed to module_HYDRO_io.F to and reads 'reservoir_type' from the reservoir parameter file when it is true. The RFC Forecasts sub-module will also instantiate and point to a levelpool reservoir that will run simultaneously as RFC forecast time series files are ingested.

ISSUE: Fixes #411 

TESTS CONDUCTED: Tested on small domain. Nothing is answer changing at this point. This only supports the next phase of the structure of the RFC Forecasts module. 

### Checklist

 - [X ] Closes issue #411 
 - [x] Tests added (unit tests and/or regression/integration tests)
 - [X ] Backwards compatible
 - [x] Requires new files? Requires new Reservoir Parameter File
 - [X ] Fully documented
 - [ ] Short description in the Development section of `NEWS.md`

### CODE REVIEW

File: trunk/NDHMS/Data_Rec/module_namelist.F [hunk 1]
```diff
@@ -55,11 +55,13 @@ subroutine read_rt_nlst(nlst)
...
           character(len=256) :: route_direction_f=""
           character(len=256) :: route_order_f=""
-          logical            :: reservoir_persistence
+          logical            :: reservoir_persistence_usgs
           character(len=256) :: reservoir_parameter_file=""
           character(len=256) :: reservoir_timeslice_path = ""
           integer            :: reservoir_observation_lookback_hours
           integer            :: reservoir_observation_update_time_interval_seconds
+          logical            :: reservoir_rfc_forecasts
+          logical            :: reservoir_not_levelpool
```
Comment (kafitzgerald) [inline trunk/NDHMS/Data_Rec/module_namelist.F]: I'm a bit out of the loop with respect to this development, but this doesn't appear to be in the documentation and it's not clear to me what it does.

File: trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/module_rfc_forecasts_properties.F [hunk 2]
```diff
@@ -6,12 +6,14 @@
...
 
 module module_rfc_forecasts_properties
+
     use module_reservoir, only: reservoir_properties
     implicit none
 
     ! Extend/derive rfc forecasts properties from the abstract base
     ! type for reservoir properties.
     type, extends(reservoir_properties) :: rfc_forecasts_properties_interface
+        integer :: lake_number                  ! lake number
```
Comment (kafitzgerald) [inline trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/module_rfc_forecasts_properties.F]: Again, perhaps just a lack of perspective, but it's not clear to me what `lake_number` is.  Is this different from `lake_id`?  A replacement?

Comment (jdmattern-noaa) [reply trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/module_rfc_forecasts_properties.F]: lake_number is lake_id. We wanted to stay consistent with the original name of 'lake_number' in the original levelpool code.

File: trunk/NDHMS/Data_Rec/module_namelist.F (hunk 1 above)
Comment (jdmattern-noaa) [reply trunk/NDHMS/Data_Rec/module_namelist.F]: reservoir_rfc_forecasts is a boolean flag to turn on/off the RFC Forecasts module. reservoir_not_levelpool is another boolean flag that is set to true if either USGS Persistence, Army Corps Persistence, or RFC Forecasts are set to true which then dictates reservoir_type to be read from a reservoir parameter file. I can add the documentation for these two variables to RFC_Forecasts_Module_README.md right now if you want or add that in the next PR.

Comment (rcabell) [reply trunk/NDHMS/Data_Rec/module_namelist.F]: The name `reservoir_not_levelpool` is awkward; could we rename it something like `reservoir_type_specified` etc. ?

Comment (jdmattern-noaa) [reply trunk/NDHMS/Data_Rec/module_namelist.F]: Good point. I will rename that now along with document these two and push in a few minutes. @jdmattern-noaa fixing.

Comment (jdmattern-noaa) [reply trunk/NDHMS/Data_Rec/module_namelist.F]: I pushed these changes.

File: trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/module_rfc_forecasts_properties.F (hunk 2 above)
Comment (kafitzgerald) [reply trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/module_rfc_forecasts_properties.F]: Ok, I see.

**Pull request merged**